### PR TITLE
Cleanup Docs role handbook

### PR DESCRIPTION
**Pull request by reylejano**
#### What type of PR is this:
/kind cleanup

#### What this PR does / why we need it:
Cleanup the Docs handbook. The k/website repo uses the `main` branch instead of `master` and adds a note regarding access to the website milestone maintainers link in issue https://github.com/kubernetes/sig-release/issues/1658

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/sig-release/issues/1658

#### Special notes for your reviewer:

/sig release
/sig docs

### CODE REVIEW

File: release-team/role-handbooks/docs/README.md
Comment (sftim) [inline release-team/role-handbooks/docs/README.md:67]: Maybe mention that you have to be an org member to even see https://github.com/orgs/kubernetes/teams/website-milestone-maintainers/

Comment (reylejano) [reply release-team/role-handbooks/docs/README.md]: Thank you for the clarification. Updated the PR

**Review by sftim — COMMENTED** [review]
LGTM

**Review by saschagrunert — APPROVED** [review]

**Comment by saschagrunert** [comment]
/hold
for @PI-Victor 
feel free to lift

**Comment by PI-Victor** [comment]
thank you!
/hold cancel

**Pull request merged**
